db/migrate/20230504001141_public_images.rb

```diff
@@ -0,0 +1,40 @@
+require 'aws-sdk-s3'
+
+class PublicImages < ActiveRecord::Migration[7.0]
+  def up
+    return unless s3_bucket
+
+    check_connection!
+    set_objects_to_readable
+
+    ActiveStorage::Blob.
+      where(service_name: "amazon").
+      where("content_type LIKE 'image/%'").
+      update_all(service_name: "amazon_public")
```

Comment:
We could have private images. The TermsOfServiceFile doesn't validate the file type. But that's really public, too. The alternative to go through all our models is more work, joining and `where(active_storage_attachments: { record_type: ["Spree::Asset", "EnterpriseGroup", "Enterprise"] }`. Enterprises also have terms and conditions but they are restricted to PDF. So at this point in time we are good here.

Reply:
Yeah I had a look at that alternative and it didn't seem viable purely due to the scale in production. You'd have to fetch each individual image and make an individual API request. I think it wouldn't work nicely.

As a side note, if we're doing this migration it probably needs to hit production before the background reports feature, which would start using `ActiveStorage` for things that aren't public. If this goes in first I think it'll be fine because all the new reports stuff will be using the non-public config and they'll get their ACLs set to private by default during upload.

Reply:
I see. It's still difficult to guess a report file key but yes, it would be good to do this migration first. I can see that this PR is still in draft mode. Is it ready from your perspective?

Reply:
Sure :+1: